src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -416,37 +433,46 @@ private void ProcessProjectChange(Solution solution, ProjectId projectId)
                 return;
             }
 
-            installedPackages = new Dictionary<string, string>();
+            var installedPackages = new Dictionary<string, string>();
+            var isEnabled = false;
 
             // Calling into NuGet.  Assume they may fail for any reason.
             try
             {
                 var installedPackageMetadata = _packageServices.GetInstalledPackages(dteProject);
                 installedPackages.AddRange(installedPackageMetadata.Select(m => new KeyValuePair<string, string>(m.Id, m.VersionString)));
+                isEnabled = true;
+            }
+            catch (ArgumentException)
+            {
+                // Nuget may throw an ArgumentException when there is something about the project 
+                // they do not like/support.
             }
             catch (Exception e) when (FatalError.ReportWithoutCrash(e))
             {
             }
 
-            _projectToInstalledPackageAndVersion.AddOrUpdate(projectId, installedPackages, (_1, _2) => installedPackages);
+            var state = new ProjectState(isEnabled, installedPackages);
+            _projectToInstalledPackageAndVersion.AddOrUpdate(
+                projectId, state, (_1, _2) => state);
         }
 
         public bool IsInstalled(Workspace workspace, ProjectId projectId, string packageName)
         {
             ThisCanBeCalledOnAnyThread();
             return _projectToInstalledPackageAndVersion.TryGetValue(projectId, out var installedPackages) &&
-                installedPackages.ContainsKey(packageName);
+                installedPackages.InstalledPackageToVersion.ContainsKey(packageName);
         }
 
         public ImmutableArray<string> GetInstalledVersions(string packageName)
         {
             ThisCanBeCalledOnAnyThread();
 
             var installedVersions = new HashSet<string>();
-            foreach (var installedPackages in _projectToInstalledPackageAndVersion.Values)
+            foreach (var state in _projectToInstalledPackageAndVersion.Values)
```

Comment:
Consider using @sharwell's `Deconstruct` trick to avoid allocating the `Values` collection.